### PR TITLE
Run non-initial HTML assets through cloneForEachLocale, and also preserve their url

### PR DESCRIPTION
**Pull request by papandreou**
(Experimental)

### CODE REVIEW

**Comment by papandreou** [comment]
@Munter, I looked into this a bit, and internationalizing non-initial HTML assets seems to work.

I had to tweak the set of assets that are moved to /static by buildProduction so that those non-initial HTML, non-fragment assets have their original url preserved, and that had the consequence that web components also stayed put. Maybe that's a good thing?

**Comment by papandreou** [comment]
CI failure appears to be some unrelated flakiness with optipng.

File: lib/transforms/buildProduction.js [hunk 1]
```diff
@@ -284,12 +284,15 @@ module.exports = function (options) {
...
             .if(!options.noFileRev)
                 .moveAssetsInOrder(query.and(
-                    {
-                        isLoaded: true,
-                        isInline: false,
-                        type: query.not(['CacheManifest', 'Rss', 'Atom']),
-                        fileName: query.not(['.htaccess', 'humans.txt', 'robots.txt'])
-                    },
+                    query.or(
+                        {type: 'Html', isFragment: true, isLoaded: true},
```
Comment (Munter) [inline lib/transforms/buildProduction.js]: fileRev Html fragments? How does that work?

**Comment by Munter** [comment]
I understand the tests, but not the filerev change. I guess the previous heuristic of not filereving html with incoming anchors is failing now, since the relations are explicitly broken by keeping the href while creating new language specific ones.

Is this where we can grab into our bag of tricks and transform the `HtmlAnchor --> Html` into `HtmlAnchor --> HtmlRedirect ==> Html.en + Html.da` ?

**Comment by Munter** [comment]
Oh, I see now that `HttpRedirect` is a relation and not an asset.

This stuff breaks our model :(

**Comment by Munter** [comment]
What if we changed tactics from the apache inspired `index.da.html` to `da/index.html` and create a coherent dependency graph for each language a the server root?

**Comment by Munter** [comment]
I've tried rerunning the failing tests. They seem to fail consistently. I wonder what changed with optipng

**Comment by papandreou** [comment]
> I understand the tests, but not the filerev change. I guess the previous heuristic of not filereving html with incoming anchors is failing now, since the relations are explicitly broken by keeping the href while creating new language specific ones.

Ah, of course that's what happened. I managed to confuse myself quite a bit over why those tests broke and missed that bit.

> Is this where we can grab into our bag of tricks and transform the HtmlAnchor --> Html into HtmlAnchor --> HttpRedirect ==> Html.en + Html.da ?

Yeah, `HttpRedirect` relations are attached to whatever asset was served as the body of the 30[12] HTTP response. I guess we could use something similar here, some sort of virtual asset that holds references to all the translated files. That map won't be easy to serialize to disc, though, so it might be better if it's a "content negotiation" mode that you switch on for the whole graph, which then affects how relations are interpreted. The idea of that scares me as well, though :/

File: lib/transforms/buildProduction.js (hunk 1 above)
Comment (papandreou) [inline lib/transforms/buildProduction.js]: `isFragment` is not the same as `isInline`. Fragments are HTML assets without an `<html>` element.

**Pull request closed without merging**